lms/views/predicates/_predicates.py

```diff
@@ -0,0 +1,63 @@
+from lms.views.lti.basic_launch import ResourceLinkParam
+
+
+def get_url_configured_param(_context, request):
+    """Get the URL param if we are configured by URL alone."""
+
+    # The 'url' is a deep linked launch reading the param we sent to the LMS
+    # to store during deep linking. All other params are situation specific.
+    for param in ["url", "canvas_file", "vitalsource_book"]:
+        if param in request.params:
+            return param
+
+    return None
+
+
+def get_db_configured_param(context, request):
+    """Get the URL link id param if we have a matching DB record."""
+
+    param, _assignment = _get_db_configured_param_and_assignment(context, request)
+
+    return param
+
+
+def _get_db_configured_param_and_assignment(context, request):
+    """Get a tuple of param and assignment if we have a matching DB record."""
+
+    for param in (
+        ResourceLinkParam.LTI,
+        ResourceLinkParam.COPIED_BRIGHTSPACE,
+        ResourceLinkParam.COPIED_BLACKBOARD,
+    ):
+        # Horrible work around
+        if param == ResourceLinkParam.LTI:
+            resource_link_id = context.resource_link_id
+        else:
+            resource_link_id = context.lti_params.get(param)
+
+        if not resource_link_id:
+            continue
+
+        assigment = request.find_service(name="assignment").get_assignment(
```

Comment:
We might be making some extra queries with the same augments depending of the type of launch.

That's fine I reckon, simply slapping a `lru_cache` might create some bugs as it's something we might create during the launch (as opposed to an application instance for example)